Terminal.Gui/Core/Application.cs

```diff
@@ -639,9 +639,13 @@ static void OnUnGrabbedMouse (View view)
 		}
 
 		/// <summary>
-		/// Merely a debugging aid to see the raw mouse events
+		/// <para>
+		/// Called for new MouseEvent events before any processing is performed or
+		/// views evaluate.  Use for global mouse handling and/or debugging.
+		/// </para>
+		/// <para>Return true to suppress the MouseEvent event</para>
 		/// </summary>
-		public static Action<MouseEvent> RootMouseEvent;
+		public static Func<MouseEvent, bool> RootMouseEvent;
```

Comment:
Unfortunately, it does appear folks are using this API. 
https://cs.github.com/?scopeName=All+repos&scope=&q=%22RootMouseEvent%22+AND+%22Terminal.GUI%22+AND+NOT+repo%3Agui-cs%2FTerminal.Gui

e.g. (There are dozens):

<img width="411" alt="image" src="https://user-images.githubusercontent.com/585482/196506291-e2a59a51-6fa1-4f76-85f4-1de49fa51b94.png">

Reply:
Ok that's fair.  I guess we could have a `RootMouseEventCancellable` or a `RootMouseEvent2`?

Alternatively we could add the `Handled` to `MouseEvent`.  I have looked into it and this is my understanding:

### MouseEvent
There are 3 public structs all called `MouseEvent` but the one dealt with by the `RootMouseEvent` event is `Terminal.Gui.MouseEvent`:

https://github.com/gui-cs/Terminal.Gui/blob/da34dedaa5eddf489a7a725b0ec8ece3546be433/Terminal.Gui/Core/Event.cs#L711

- `MouseEvent` is a `struct`
- `MouseEventArgs` is a class that wraps `MouseEvent` and adds `Handled` property and inherits from `EventArgs`

https://github.com/gui-cs/Terminal.Gui/blob/df7285954210c2d64ce98e5f4043263b4c86f60a/Terminal.Gui/Core/View.cs#L2745-L2763

We could add `Handled` to `MouseEvent` struct and redirect the one in `MouseEventArgs` to point to it's wrapped instance?